patches/minecraft/net/minecraft/client/renderer/RenderType.java.patch

```diff
@@ -34,3 +34,12 @@
     }
  
     public static RenderType m_110502_() {
+@@ -375,7 +_,7 @@
+    }
+ 
+    public static List<RenderType> m_110506_() {
+-      return ImmutableList.of(m_110451_(), m_110457_(), m_110463_(), m_110466_(), m_110503_());
++      return net.minecraftforge.client.renderer.LevelRenderPhaseManager.getInstance().getKnownTypes();
```

Comment:
In my opinion this is missing an accompanying patch in `RenderBuffers`: The `RenderBuffers#fixedBuffers` map which is used to construct the `RenderBuffers#bufferSource` needs to contain all chunk layers because requesting a buffer from said buffer source for a certain `RenderType` after using a different `RenderType` that isn't in the map of fixed buffers will cause the current data in the buffer to be uploaded to the GPU because "non-fixed" `RenderType`s use a shared buffer.

While this is not an issue for the static chunk rendering, which uses the `ChunkBufferBuilderPack` to get its buffers, I would still expect a `RenderType`, which has a pre-allocated buffer and can be used in static chunk rendering, to be useable in a dynamic context (i.e. `EntityRenderer` or `BlockEntityRenderer`) without causing an upload to the GPU every time I switch the `RenderType` in use to render the next thing in the dynamic renderer (which kills performance when a few instances of the related `Entity` or `BlockEntity` are present).